bin/aws-configure

```diff
@@ -18,8 +18,7 @@
 
 function enforce_requirements() {
   # Make sure the AWS CLI is installed
-  which aws &>/dev/null
-  if [[ $? -ne 0 ]]; then
+  if ! which aws &>/dev/null; then
```

Comment:
You might want to use `command -v` here: https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then

Reply:
So I spent like...90 minutes yesterday on that. I can't figure out how to mock it :/ Any ideas?

Reply:
I'm not sure what you mean by "how to mock" it.
```bash
if ! command -v aws > /dev/null 2>&1; then
```
will do basically the same thing, but it won't depend on a separate executable and has the benefit of still working if you provided an `aws` shell function (if you wanted to wrap the executable or something).

Reply:
When I'm writing bats tests to test the scenario where command can and cannot find aws - how do I handle that? For example: https://github.com/chef/ci-studio-common/pull/92/files#diff-25dca930eaf78afc6d300e8a1193354dR6

Reply:
After poking around for a bit, I'd suggest removing that test and add
```bash
set -eou pipefail
```
to the top of your script. This is the [unofficial bash strict mode](http://redsymbol.net/articles/unofficial-bash-strict-mode/) and I think it makes it much easier to write good, usable bash scripts.

In this case, you can leave in the `enforce_requirements` function if you like, but since you have `set -e`, it's not really necessary. As soon as the script gets to a place where it would try to execute `aws`, it will exit with an appropriate error message like (on my system):
```
The program 'aws' is currently not installed. You can install it by typing:
sudo apt install awscli
```
If you want to have a different, specific error message, you can do that with the `enforce_requirements` code you have now, or you could take the approach of wrapping the `aws` command like so:

```bash
function aws() {
  # Make sure the AWS CLI is installed
  if command -v aws > /dev/null 2>&1; then
    command aws "$@" # run the aws binary, passing all args
  else
    echo "${0##*/}: Unable to find 'aws' executable."
    echo "Make sure you have the aws cli installed. Try 'install-tool aws'."
    exit 2
  fi
}
```

The benefit here is that you have to explicitly remember to call `enforce_requirements`, but the wrapper can't be forgotten and will only complain if `aws` needed to execute.

Since `bats` doesn't allow you to stub/mock bash functions themselves (only executables), there's only so much we can test, and I think testing the specific behavior of a missing command like `aws` doesn't have a lot of value when we already have things like `set -e`.